src/libraries/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -50,7 +50,7 @@ namespace System.Collections.Generic
     [SuppressMessage("Microsoft.Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix", Justification = "By design")]
     [Serializable]
     [System.Runtime.CompilerServices.TypeForwardedFrom("System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
-    public class HashSet<T> : ICollection<T>, ISet<T>, IReadOnlyCollection<T>, ISerializable, IDeserializationCallback
+    public class HashSet<T> : IReadOnlySet<T>, ICollection<T>, ISet<T>, IReadOnlyCollection<T>, ISerializable, IDeserializationCallback
```

Comment:
You may want to put `IReadOnlySet` towards the end of the list of interfaces. The order of the interfaces affects casting performance - the interfaces earlier in the list are faster to cast to. `IReadOnlySet` is going to be used rarely, so it is not worth optimizing for it.

Reply:
Never knew that! I'm on it. More-so trying to figure out why Immutable isn't detecting the interface. But I will get right on that

Reply:
Immutable is NuGet package that supports downlevel platforms where this interface does not exist. You will need to include this interface on Immutable for netcoreapp5.0 build only.

Reply:
@jkotas any examples on other places we're doing this?

Reply:
E.g. https://github.com/dotnet/runtime/blob/master/src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs#L122

Reply:
So to clarify, something like for the interface import?
```
public sealed partial class ImmutableHashSet<T> : IImmutableSet<T>, IHashKeyCollection<T>, IReadOnlyCollection<T>, ICollection<T>, ISet<T>, ICollection, IStrongEnumerable<T, ImmutableHashSet<T>.Enumerator>
#if NETSTANRD5_0
        IReadOnlySet<T>
#endif
```

Reply:
To make this future proof and make it build without any changes once we get to netcoreapp6.0, there is a preference to do these ifdefs via exclusion, e.g.
`#if !NETSTANDARD1_0 && !NETSTANDARD1_3 && !NETSTANDARD2_0`

Another style used for this are partial types, e.g. here: https://github.com/dotnet/runtime/tree/master/src/libraries/System.Threading.Channels/ref

Reply:
@jkotas does my ifdef for immutable hashset/sortedset make sense? I don't mean to seem dumb, but I only vaguely know what I'm doing :)

Reply:
If it makes all build flavors happy, it looks fine to me. Could you please resolve the conflicts so that the CI can run?